src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,118 @@
+
+    @Test
+    public void shouldSupportConfigurationAsCodeExwsAllocateStep() throws Exception {
+        // Test ExwsAllocateStep
+        ExwsAllocateStep.DescriptorImpl descriptor =  ExtensionList.lookupSingleton(ExwsAllocateStep.DescriptorImpl.class);
+        List<DiskPool> diskPools = descriptor.getDiskPools();
+        DiskPool diskPool = diskPools.get(0);
+
+        assertThat(diskPool.getDiskPoolId(), is("diskpool1"));
+        assertThat(diskPool.getDisplayName(), is("diskpool1 display name"));
+        assertThat(diskPool.getDisks().get(0).getDiskId(), is("disk1"));
+        assertThat(diskPool.getDisks().get(0).getDisplayName(), is("disk one display name"));
+        assertThat(diskPool.getDisks().get(0).getMasterMountPoint(), is("/tmp"));
+    }
+
+
+    @Test
+    public void shouldSupportConfigurationAsCodeExwsStep() throws Exception {
+        // Test ExwsStep
+        ExwsStep.DescriptorImpl globalTemplateDescriptor = ExtensionList.lookupSingleton(ExwsStep.DescriptorImpl.class);
+        List<Template> templates = globalTemplateDescriptor.getTemplates();
+
+        assertThat(templates.get(0).getLabel(), is("all"));
+        assertThat(templates.get(0).getNodeDiskPools().get(0).getDiskPoolRefId(), is("dp1"));
+        NodeDisk nodeDisk = templates.get(0).getNodeDiskPools().get(0).getNodeDisks().get(0);
+        assertThat(nodeDisk.getDiskRefId(), is("dp1refid1"));
+        assertThat(nodeDisk.getNodeMountPoint(), is("/tmp/template11"));
+        nodeDisk = templates.get(0).getNodeDiskPools().get(0).getNodeDisks().get(1);
+        assertThat(nodeDisk.getDiskRefId(), is("dp1refid2"));
+        assertThat(nodeDisk.getNodeMountPoint(), is("/tmp/template12"));
+        assertThat(templates.get(0).getNodeDiskPools().get(1).getDiskPoolRefId(), is("dp2"));
+        nodeDisk = templates.get(0).getNodeDiskPools().get(1).getNodeDisks().get(0);
+        assertThat(nodeDisk.getDiskRefId(), is("dp2refid1"));
+        assertThat(nodeDisk.getNodeMountPoint(), is("/tmp/template21"));
+    }
+
+    @Test
+    public void shouldSupportConfigurationAsCodeMasterProperty() throws Exception {
+        Computer computer = j.getInstance().getComputer("");
+        Node node = computer.getNode();
+        List<NodeDiskPool> nodeDiskPools = node.getNodeProperties().get(ExternalWorkspaceProperty.class).getNodeDiskPools();
+        assertThat(nodeDiskPools.get(0).getDiskPoolRefId(), is("master-node-id"));
+        assertThat(nodeDiskPools.get(0).getNodeDisks().get(0).getDiskRefId(), is("master-node-disk"));
+        assertThat(nodeDiskPools.get(0).getNodeDisks().get(0).getNodeMountPoint(), is("/tmp/master-node"));
+    }
+
+    @Test
+    public void shouldSupportConfigurationAsCodeAgentProperty() throws Exception {
+        Computer computer = j.getInstance().getComputer("");
+        Node node = computer.getNode();
+        List<NodeDiskPool> nodeDiskPools = node.getNodeProperties().get(ExternalWorkspaceProperty.class).getNodeDiskPools();
```

Comment:
I think we need some assertions for this test, don't we?

Reply:
It seems this way to me too. I will look into this.